app/c2/gist.py

```diff
@@ -0,0 +1,138 @@
+import json
+import uuid
+import aiohttp
+import re
+from base64 import b64encode
+
+from app.objects.c_c2 import C2
+
+
+class Gist(C2):
+
+    def __init__(self, module_info):
+        self.name = 'Gist'
+        self.key = module_info['config']['key']
+        self.c2_type = module_info['c2_type']
+
+    def encode_config_info(self):
+        """
+        Returns one of the GIST API keys to be encoded into the agent
+        :return: GIST api key
+        """
+        return self.key
+
+    async def get_results(self):
+        """
+        Retrieve all GIST posted results for a particular api key
+        :return:
+        """
+        results = await self._get_raw_gist_urls(comm_type='results')
+        result_content = await self._get_gist_content([result[0] for result in results])
+        await self._delete_gists([result[1] for result in results])
+        return result_content
+
+    async def get_beacons(self):
+        """ Retrieve all GIST beacons for a particular api key"""
```

Comment:
Missing return value in doctoring

Reply:
Also - this says "for a particular API key"... is the plan to eventually pass through the API key as an argument here?